arte-downloader.user.js

```diff
@@ -2,163 +2,267 @@
+
+    return button;
+}
 
-  var button = document.createElement('a');
-  button.setAttribute('class', 'btn btn-default');
-  button.setAttribute('style', 'text-align: center; display: table-cell;');
-  button.innerHTML= "Download <strong>Metadata</strong> <span class='icomoon-angle-right pull-right'></span>";
-  
-  // Get the content of the JSON file.
-  var jsonUrl = getJsonUrl(element);
-  console.log(jsonUrl);
-  GM_xmlhttpRequest({
-    method: "GET",
-    url: jsonUrl,
-    onload: function(response) {
-      var metadata = getMetadata(response);
-      // Properly encode to Base 64.
-      var encodedData = window.btoa(unescape(encodeURIComponent(metadata)));
-      // The href will output a text file. 
-      // For a CSV file, that would be: data:application/octet-stream,field1%2Cfield2%0Afoo%2Cbar%0Agoo%2Cgai%0A
-      button.setAttribute('href', 'data:application/octet-stream;charset=utf-8;base64,' + encodedData);
+function createLanguageComboBox() {
+    var languageComboBox = document.createElement('select');
+    languageComboBox.setAttribute("id", "languageComboBox");
+    for (l in availableLanguages) {
+        if (availableLanguages[l] !== 0) {
+            languageComboBox.innerHTML += "<option value='" + l + "' selected>" + availableLanguages[l] + "</option>";
+        }
     }
-  });
-  return button;
+    languageComboBox.setAttribute('class', 'btn btn-default');
+    languageComboBox.setAttribute('style', 'width:97%');
+
+    return languageComboBox;
 }
 
-/*
- * Run an X-Path query to retrieve the URL of the JSON file which contains the MP4 video URLs.
- */
-function getJsonUrl(element) {
+function createButtons(videoElement) {
+    preParsePlayerJson();
+
+    var parent = videoElement.parentNode.parentNode;
+    var container = document.createElement('div');
+    container.setAttribute('style', 'display: table; width: 100%;');
+    parent.appendChild(container);
 
-  // Get the value of the "arte_vp_url" attribute which contains the player URL.
-  // playerUrl = result.snapshotItem(0).getAttribute("arte_vp_url");
-  var playerUrl = element.getAttribute("arte_vp_url");
+    container.appendChild(createLanguageComboBox());
+    var languageComboBox = document.getElementById("languageComboBox");
+    var selectedLanguage = languageComboBox.options[languageComboBox.selectedIndex].value;
 
-  // Get the URL of the JSON file by removing the "player/".
-  var json = playerUrl.replace("player/", "");
+    if (bLive) {
+        container.appendChild(createButton('High', selectedLanguage, "RPM4 playlist (copy/paste into VLC) <span class='icomoon-angle-down force-icomoon-font'></span>"));
+    } else {
+        container.appendChild(createButton('Low', selectedLanguage));
+        container.appendChild(createButton('Standard', selectedLanguage));
+        container.appendChild(createButton('High', selectedLanguage));
+        //container.appendChild(createButtonMetadata(videoElement)); // @TODO display instead of download
```

Comment:
Coucou!
since i don't see the point in downloading a file with metadata (because the .mp4 have it)
i propose to display the metadata in a box when clicking on the link.